frontend/components/cards/ModuleUnionCard.tsx

```diff
@@ -48,7 +73,9 @@ const ModuleUnionCard: FC<ModuleUnionCardProps> = ({
     case "entities.organization":
     case "entities.entity":
     case "entities.group":
-      content = <div dangerouslySetInnerHTML={{ __html: module.name }} />;
+      content = highlightSnippet && ( // TODO: Replace Regex with more formal fix
+        <HighlightEllipsis unsafeHTML={highlightSnippet} />
+      );
```

Comment:
How about adding a fallback value in case there's no `highlightSnippet`?

Something like...
```
      content = highlightSnippet && (
        <HighlightEllipsis unsafeHTML={highlightSnippet} />
      ) || (
        <HTMLEllipsis unsafeHTML={entity.description} maxLine="4" basedOn="words" />
      )
```

Reply:
Added fallbacks to all content types.